File: go/bubbleSort.go
package main

import "fmt"

func main() {
	array := []int{100, 50, 30, 700, 500, 350, 10, 900, 20}
	fmt.Println("before")
	fmt.Println(array)
	for i := 0; i < len(array); i++ {
		for j := 0; j < len(array)-1-i; j++ {
			if array[j] > array[j+1] {
				array[j], array[j+1] = array[j+1], array[j]
			}
		}
	}
	fmt.Println("after")
	fmt.Println(array)
}

File: java/BubbleSort.java
public class BubbleSort {
	public static void main(String[] args) {
	int array[] = {30, 10, 20, 1, 100, 700, 500, 50};
	System.out.println("before");
	for (int a:array)
		System.out.println(a);
	for (int i = 0; i < array.length - 1; i++) {
		for (int j = array.length - 1; j > i; j--) {
			if (array[j - 1] > array[j]) {
				int tmp = array[j - 1];
				array[j - 1] = array[j];
				array[j] = tmp;
			}
		}
	}
	System.out.println("after");
	for (int a:array)
		System.out.println(a);
	}
}
